### PR TITLE
test: add unit-tests for middleware modules

### PR DESCRIPTION
**Pull request by pnts-se**
Will change typescript config to use projects.
That way we do not build `/__tests__` and `/__mocks__`, only `/src` and whatever is imported from there.

Add good unit-test coverage of middleware modules.

image

### CODE REVIEW

File: __mocks__/ConsumerMock.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default class ProducerMock {
+	id = 'id';
+
+	resume = jest.fn();
+	pause = jest.fn();
+	setPreferredLayers = jest.fn();
+	setPriority = jest.fn();
```
Comment (havfo) [inline __mocks__/ConsumerMock.ts:7]: A producer does not have this, and the following method, but doesn't matter

Comment (havfo) [reply __mocks__/ConsumerMock.ts:7]: Hmmm...producer mock in the wrong file?

**Review by havfo — APPROVED** [review]
LGTM, awesome!

File: __mocks__/ConsumerMock.ts (hunk 1 above)
Comment (pnts-se) [reply __mocks__/ConsumerMock.ts:7]: I think I copied ProducerMock.ts when I created ConsumerMock.ts and forgot to change the class name. The way I import it in mediaMiddleware.test.ts makes it work anyway and I didn't notice. I'll fix it before merging.

**Pull request merged**